wave_lang/kernel/wave/execution_engine/register_dialects.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright 2025 The Wave Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <mlir/Dialect/LLVMIR/LLVMDialect.h>
+#include <mlir/IR/DialectRegistry.h>
+#include <mlir/Target/LLVMIR/Dialect/Builtin/BuiltinToLLVMIRTranslation.h>
+#include <mlir/Target/LLVMIR/Dialect/LLVMIR/LLVMToLLVMIRTranslation.h>
+
+mlir::DialectRegistry createMLIRContextRegistry() {
+  mlir::DialectRegistry registry;
+  registry.insert<mlir::LLVM::LLVMDialect>();
```

Comment:
Do we need something like AMDGPU as well?

Reply:
See my prev comment on O-level, this pipeline will only for the host x86 code.